### DISCUSSION TITLE
Nightscout on google cloud: Two instances after change hostname

### DISCUSSION THREAD

**Discussion by floheigl** [NightscoutFoundation/xDrip#3212] [Q&A] [answered]
Hi,
recently switched to NS on google cloud! Great work on the "non-tech-people-proof" menu solution!

I have had my initial setup working like a charm for about a month now, but since my first choice of domain was blocked at work, I decided to change it.
I followed this guide from the FAQs:
https://navid200.github.io/xDrip/docs/Nightscout/ChangeHostname.html

On the first try i picked a subdomain on mooo.com which then didn't really work (didn't respond to ping). So i tried a second time with raspberryip.com.

![ns2](https://github.com/NightscoutFoundation/xDrip/assets/28005934/474e1e00-935a-49d5-87b9-c663c237fc43)
(Image: Original setup Nov 4th, First try of URL-Change Nov 27 9:33, 2nd try: 9:42)

Now, **both** the original one from my first setup and the new one on raspberryip.com are up and running. They also both get fresh data every 5 minutes even though xDrip obviously only uploads to one URL (pointing to same DB?). They are not the **same instance though**, as the "Custom title" and the shown plugins are different.

here's my status page:
![ns1](https://github.com/NightscoutFoundation/xDrip/assets/28005934/09c2dae0-c281-4f67-b482-aa03579a6e38)

Question: 
- Do I really have two instances of NS running on the same VM?
- If yes: How do i remove the old one?
- Should this be addressed in the GUIDE?

**Comment by floheigl** [comment]
OK,
I just realized that the "Custom title" and "Plugins shown" are probably stored in something like a cookie. So my assumption that i have two different instances running was probably wrong.
Would really appreciate if someone could confirm or show me a definitive way to rule out some freaky double-instance state.

In case this was all confusion on my part:
So should the guide be renamed to "Adding Hostname"? Or maybe include a reminder for removing the old subdoman in FreeDNS?

**Comment by Navid200** [comment, marked as answer]
To see how many virtual machines you are running on Google Cloud, go to compute engine:  https://navid200.github.io/xDrip/docs/Nightscout/ComputeEngine.html
On that page, you can see how many instances you have running.
Even if you satisfy all the free tier requirements, you will be charged if you have more than one instance.

FreeDNS maps an ip address to a hostname.
It is quite possible that you have 2 hostnames in FreeDNS that both are mapping the same IP to two different hostnames.
If that's the case, both hostnames will access the same IP address (same virtual machine).  If this is the case, your uploader is uploading to a hostname, which is mapped to your IP address.
Then, when you access the other hostname in a browser or a follower, you are also seeing the correct values.  
This only means that you have two different hostnames mapped to the same IP address.

The latter scenario is not necessarily a problem.
You can verify it by logging into your FreeDNS account and going to the subdomains page.  You can see the IP addresses of each subdomain.
If this is the case, and you are not happy, you can remove one of the two subdomains as long as it is not the one your virtual machine is using.

To see which subdomain the virtual machine is using, use this: https://navid200.github.io/xDrip/docs/Nightscout/Hostname.html  

Please let me know if you have any questions.

**Reply by floheigl** [reply]
Thanks for your answer! 
I didn't think of looking up the IP-Adresses pointed to in freedns. They are indeed the same.

Would you consider adding your Tip to the guide?: "...  you can remove one of the two subdomains as long as it is not the one your virtual machine is using."

**Reply by Navid200** [reply]
Creating a guide is just a part of the task.  Maintaining it and keeping it up-to-date is the real challenge.
The more data you add to a guide, the more susceptible it becomes to containing data that is outdated sooner or later.  

If you ask yourself "OK, what else can I write about to help users"?  There will be no end to what you can write.  And someone is going to find it useful.  But, at some point, you have to draw the line and say OK, this is the guide, I will take care of the rest in the forums.
